core/shared/src/main/scala/cats/effect/LiftIO.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect
+
+import cats.{~>, Applicative, Functor}
+import cats.data.{EitherT, IorT, Kleisli, OptionT, ReaderWriterStateT, StateT, WriterT}
+import cats.kernel.Monoid
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("""Cannot find implicit value for LiftIO[${F}].
+Building this implicit value might depend on having an implicit
+s.c.ExecutionContext in scope, a Scheduler or some equivalent type.""")
+trait LiftIO[F[_]] {
+  def liftIO[A](ioa: IO[A]): F[A]
+}
+
+object LiftIO {
+
+  def apply[F[_]](implicit F: LiftIO[F]): F.type = F
+
+  /**
+   * [[LiftIO.liftIO]] as a natural transformation
+   */
+  def liftK[F[_]: LiftIO]: IO ~> F = λ[IO ~> F](LiftIO[F].liftIO(_))
```

Comment:
> A little help with the dotty compilation issue... Please.

The issue is that kind-projector doesn't exist on Dotty! I suppose I should technically remove it from the main project too, but I've been lazy.

The rewrite is pretty trivial though.

```suggestion
  def liftK[F[_]: LiftIO]: IO ~> F = 
    new (IO ~> F) { def apply[A](fa: F[A]) = LiftIO[F].liftIO(fa) }
```

It's rather annoying that we can't use type lambdas, but until Dotty improves its support for higher-rank type inference, we can't even shim it. :-(

Reply:
Yeah, this looks reasonable. I don't follow Dotty much, sorry.

Reply:
No worries! We aren't using any of the fancy stuff, because we're cross building. So honestly it's easiest to think of Dotty in our case as "Scala with way fewer bugs and no macros or plugins".

Reply:
@djspiewak FYI, Dotty supports `-Ykind-projector` with some limitations

Reply:
@mpilquist Yeah, we're using it, but `-Ykind-projector` doesn't support the value-level aspects of kind-projector. I spent a bit of time trying to build a Dotty-specific shim using higher-rank types to get around this, but unfortunately it appears that higher-rank types don't currently infer at all in Dotty (@smarter wants to work on this when he gets time), so we're stuck using anonymous inner classes everywhere.